bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/discovery/BridgeDiscoveryParticipant.java

```diff
@@ -0,0 +1,152 @@
+    private PublicInformation bridgeInformation = new PublicInformation();
+
+    public BridgeDiscoveryParticipant() {
+        // create http client upfront to later request public information from SHC
+        SslContextFactory sslContextFactory = new SslContextFactory.Client.Client(true); // Accept all certificates
+        sslContextFactory.setTrustAll(true);
+        sslContextFactory.setValidateCerts(false);
+        sslContextFactory.setValidatePeerCerts(false);
+        sslContextFactory.setEndpointIdentificationAlgorithm(null);
+        httpClient = new HttpClient(sslContextFactory);
+    }
+
+    protected BridgeDiscoveryParticipant(HttpClient customHttpClient) {
+        httpClient = customHttpClient;
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return SUPPORTED_THING_TYPES_UIDS;
+    }
+
+    @Override
+    public String getServiceType() {
+        return "_http._tcp.local.";
+    }
+
+    @Override
+    public @Nullable DiscoveryResult createResult(ServiceInfo serviceInfo) {
+        logger.trace("Bridge Discovery started for {}", serviceInfo);
+
+        @Nullable
+        final ThingUID uid = getThingUID(serviceInfo);
+        if (uid == null) {
+            return null;
+        }
+
+        logger.trace("Discovered Bosch Smart Home Controller at {}", bridgeInformation.shcIpAddress);
+
+        return DiscoveryResultBuilder.create(uid)
+                .withLabel("Bosch Smart Home Controller (" + bridgeInformation.shcIpAddress + ")")
+                .withProperty("ipAddress", bridgeInformation.shcIpAddress)
+                .withProperty("shcGeneration", bridgeInformation.shcGeneration)
+                .withProperty("apiVersions", bridgeInformation.apiVersions).withTTL(60 * 60 * 24).build();
+    }
+
+    @Override
+    public @Nullable ThingUID getThingUID(ServiceInfo serviceInfo) {
+        String ipAddress = discoverBridge(serviceInfo).shcIpAddress;
+        if (!ipAddress.isBlank()) {
+            return new ThingUID(BoschSHCBindingConstants.THING_TYPE_SHC, ipAddress.replace('.', '-'));
+        }
+        return null;
+    }
+
+    protected PublicInformation discoverBridge(ServiceInfo serviceInfo) {
+        logger.trace("Discovering serviceInfo {}", serviceInfo);
+
+        if (serviceInfo.getHostAddresses() != null && serviceInfo.getHostAddresses().length > 0
+                && !serviceInfo.getHostAddresses()[0].isEmpty()) {
+            String address = serviceInfo.getHostAddresses()[0];
+            logger.trace("Discovering InetAddress {}", address);
+            // store all information for later access
+            bridgeInformation = getPublicInformationFromPossibleBridgeAddress(address);
+        }
+
+        return bridgeInformation;
+    }
+
+    protected PublicInformation getPublicInformationFromPossibleBridgeAddress(String ipAddress) {
+
+        String url = String.format("https://%s:8446/smarthome/public/information", ipAddress);
```

Comment:
We already have a method in `BoschHttpClient` that returns this. If possible we should re-use it.